javascript-algorithms-and-data-structures/learn-basic-string-and-array-methods-by-building-a-music-playlist-app/scripts.js

```diff
@@ -0,0 +1,202 @@
+    { title: "Chasing That Feeling", cover: "https://cdn.freecodecamp.org/MP3%20Player%20Project/Quincy%20Larson%20Album%20Art.jpg", artist: "Quincy Larson", duration: "2:43" , src:"https://cdn.freecodecamp.org/MP3%20Player%20Project/Chasing%20That%20Feeling.mp3"},
+  ];
+
+  // song playlist
+const songListDiv = document.getElementById("playlist__songs");
+const songPlaylistDiv = document.querySelector(".playlist");
+
+function renderSongs(songs) {
+
+// map Array Method
+// TODO: First Array for steps
+const songList = songs.map((song) => {
+    return `
+    <div class="playlist__song" role="button" tabindex="0">
+        <div class="playlist__song-info">
+            <p class="playlist__song-title">${song.title}</p>
+            <p class="playlist__song-artist">${song.artist}</p>
+        </div>
+        <div class="playlist__song-rating">
+
+        </div>
+        <div class="playlist__song-duration">
+            <p>${song.duration}</p>
+        </div>
+    </div>
+    `;
+    }).join('');
+
+    songListDiv.innerHTML = songList;
+}
+
+renderSongs(songs);
+
+
+// audio API
+const audio = new Audio();
+
+let currentIndex = 0;
+updatePlayingSongBkg(currentIndex);
+
+const playPausebtn = document.querySelector(".play");
+
+function playSong(song) {
+    const oldIndex = currentIndex;
+    // using indesOf method
+    const newIndex = songs.indexOf(song);
+
+    playPausebtn.innerHTML = '<img src="assets/pause.svg" alt="pause button" />'
+
+    if (oldIndex !== newIndex) {
+        clearSongBgs();
+        updatePlayingSongBkg(newIndex)
+    }
+
+    audio.src = song.src;
+    document.getElementById("player__song-title").innerText = song.title;
+    document.getElementById("player__song-artist").innerText = song.artist;
+    document.getElementById("player__album-art").innerHTML = `<img src="${song.cover}" alt="song cover art" />`;
+    audio.play();
+    currentIndex = newIndex;
+
+    updatePlayingSongBkg(newIndex);
+
+}
+
+function pauseSong() {
+  audio.pause();
+  playPausebtn.innerHTML = '<img src="assets/play.svg" alt="play button" />'
+}
+
+function togglePlayPause() {
+  if (audio.paused) {
+    playSong(songs[currentIndex]);
+  } else {
+    pauseSong();
+  }
+
+  updatePlayingSongBkg(currentIndex)
+
+}
+
+function toggleClosePlaylist() {
+    if (songListDiv.style.visibility == "hidden") {
+        songListDiv.style.visibility = "visible";
+        songListDiv.style.display = "flex";
+
+    } else {
+        songListDiv.style.visibility = "hidden";
+        songListDiv.style.display = "none";
+    }
+}
+
+function currentSong() {
+  return songs.find((song) => {
+    song.src === audio.src
+}) || songs[currentIndex];
+}
+
+// player controls
+function nextSong() {
+  const nextIndex = (currentIndex + 1) % songs.length;
+  playSong(songs[nextIndex]);
+  currentIndex = nextIndex;
+}
```

Comment:
```suggestion

    if (document.activeElement.classList.contains('playlist__song')) {
        const songDivs = document.querySelectorAll(".playlist__song");
        songDivs[currentIndex].focus();
    }
}
```

This keeps the keyboard focus indicator in sync with the arrow keys/selected indicator. Right now it is possible to have the focus indicator on a different song than the currently selected song if the user tabs into the playlist. If the keyboard focus is currently on a song in the playlist and the arrow keys are used then the keyboard focus indicator will jump to the currently selected song.